src/ophyd_async/epics/signal/_aioca.py

```diff
@@ -228,7 +228,18 @@ def make_converter(
     else:
         value = list(values.values())[0]
         # Done the dbr check, so enough to check one of the values
-        if datatype and not isinstance(value, datatype):
+        if (
+            datatype
+            and not isinstance(value, datatype)
+            and not (
+                datatype is int
+                and isinstance(value, float)
+                and (
+                    get_unique({k: v.precision for k, v in values.items()}, "precision")
+                    == 0
```

Comment:
Seperate this `bool` out into a local variable `is_prec_zero_float`, moving the comment to where it's defined above